### DISCUSSION TITLE
playbook plugins

### DISCUSSION THREAD

**Discussion by Willyfrog** [mattermost/community#78] [Moonshot Ideas]
a bit related to https://github.com/mattermost/community/discussions/77

Have plugins related to especific playbooks, so they react to events in the playbook and in the channel a playbook is being run.

In the end a playbook is a way to automate human processes, so having ways to modify that behaviour to customer specific needs and/or given specific conditions would be great.

P.S.: we need that tournament plugin so this is needed! 😆

**Comment by larkox** [comment]
I have been cheering for apps bindings and hooks for focalboard and Incident Response, so this would be a great start for that.

**Reply by Willyfrog** [reply]
this might be the "serious" application you mentioned in the other discussion ;)

**Comment by chetanyakan** [comment]
Also, a bit unrealistic idea could be to have bot users parse natural language (NLP) and create playbooks.

**Reply by Willyfrog** [reply]
that would move us to the moonshot ideas! but yeah, totally cool! :D

**Comment by itao** [comment]
Nice! This is exactly the direction playbooks are headed with the triggers and actions capability. Today, we hardcode common triggers in UI to make it easy to configure actions, for example:
- when a run starts
  - create a public or private channel
  - send announcement message in another channel
  - add members to the channel
- when a member joins the channel
  - send them a configurable welcome message
  - categorize channel
- when a run finishes
  - archive channel
  - export channel messages

The goal is to eventually expose a full Events API so apps can subscribe to the trigger they care about, whether it's a checklist item marked as done or someone new took on the owner role or anything inbetween.

**Reply by esethna** [reply]
Awesome!
